packages/core/src/types.ts

```diff
@@ -120,24 +122,34 @@ export interface RemoteRoot<
     ...rest: IfAllOptionalKeys<
       PropsForRemoteComponent<Type>,
       [
-        PropsForRemoteComponent<Type>?,
-        Iterable<
+        (PropsForRemoteComponent<Type> | null)?,
+        MaybeArray<
```

Comment:
Can we set the type so that people know to use either the old way or the new way?

```ts
root.createComponent('Stack', undefined, [root.createComponent('Text', undefined, 'This will be fun!')])
```

or 

```ts
root.createComponent('Stack', undefined, root.createComponent('Text', undefined, 'This will be fun!'), ...)
```
Maybe something like:

```ts
[
   (PropsForRemoteComponent<Type> | null)?, 
   [AllowedChildren<
       AllowedChildrenTypes,
        RemoteRoot<AllowedComponents, AllowedChildrenTypes>,
        true
     >]?,
] | 
[
  (PropsForRemoteComponent<Type> | null)?, 
   ...AllowedChildren<
       AllowedChildrenTypes,
       RemoteRoot<AllowedComponents, AllowedChildrenTypes>,
       true
    >[]
]
```

Reply:
I think this type is correct — this part of the tuple just covers the third positional argument, which can either be an `AllowedChildren<...>` or nothing (hence the `?`). The 4th..nth positional arguments are covered by the "rest" part of the tuple definition (starting with `...` below)

Reply:
I'm not sure I'm just reading it incorrectly but the current typing does not show an error when you do this:

```ts
root.createComponent('Card', {title: 'a'}, ['1', '2'], '3');
```
If we did the following, I think it will make it so that only either an array of children or positional arguments  is correct:

```ts
createComponent<Type extends AllowedComponents>(
    type: Type,
    ...rest: IfAllOptionalKeys<
      PropsForRemoteComponent<Type>,
      | [
          (PropsForRemoteComponent<Type> | null)?,
          AllowedChildren<
            AllowedChildrenTypes,
            RemoteRoot<AllowedComponents, AllowedChildrenTypes>,
            true
          >[]?,
        ]
      | [
          (PropsForRemoteComponent<Type> | null)?,
          ...AllowedChildren<
            AllowedChildrenTypes,
            RemoteRoot<AllowedComponents, AllowedChildrenTypes>,
            true
          >[]
        ]
    >
  )
```

Reply:
Thanks, I think I see what you mean now. Can you check to see if the new type for this is what you are expecting?